### PR TITLE
Break mz into modules, remove unused bluebird reference

### PR DESCRIPTION
**Pull request by rbuckton**
This PR improves the existing type definition for `mz` for types-2.0 by switching from ambient module declarations to external modules. This fixes an issue where a referring project must add a `"types"` entry into their `tsconfig.json` to reference a module such as `mz/fs`.

This also removes an unused reference to the type definitions for `bluebird`, and updates the various exported functions to be more consistent with recent changes to `node.d.ts`.

### CODE REVIEW

**Comment by rbuckton** [comment]
// CC: @RyanCavanaugh, @vladima

File: mz/index.d.ts [hunk 1]
```diff
@@ -5,793 +5,12 @@
...
-declare module "mz"{
-    export import child_process = require("mz/child_process");
-    export import crypto = require("mz/child_process")
-    export import dns = require("mz/dns")
-    export import fs = require("mz/fs")
-    export import readline = require("mz/readline")
-    export import zlib = require("mz/zlib")
-}
+import * as child_process from "mz/child_process";
+import * as crypto from "mz/child_process";
```
Comment (bman61) [inline mz/index.d.ts]: should this be: import * as crypto from "mz/crypto";

Comment (rbuckton) [inline mz/index.d.ts]: Good catch. It looks like this was incorrect in the original file as well.

**Comment by rbuckton** [comment]
// CC: @mhegazy

**Pull request merged**